### PR TITLE
Track auto-provisioned queued provisioning node groups.

### PR DESCRIPTION
**Pull request by dsafdsa1**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
It allows us to differentiate between standard and queued-provisioning node groups that are getting auto-provisioned.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

File: cluster-autoscaler/metrics/metrics.go [hunk 1]
```diff
@@ -390,20 +391,22 @@ var (
...
 	)
 
-	nodeGroupCreationCount = k8smetrics.NewCounter(
+	nodeGroupCreationCount = k8smetrics.NewCounterVec(
 		&k8smetrics.CounterOpts{
 			Namespace: caNamespace,
 			Name:      "created_node_groups_total",
 			Help:      "Number of node groups created by Node Autoprovisioning.",
 		},
+		[]string{"queued_provisioning"},
```
Comment (towca) [inline cluster-autoscaler/metrics/metrics.go:400]: IIRC `queued_provisioning` is a GCE/GKE-specific concept, right? These metrics are supposed to be cloud-provider-agnostic. WDYT about a type/kind (or other appropriate more generic term) label instead? And then cloud-provider-specific code can set the value to `queued_provisioning`?

Comment (dsafdsa1) [reply cluster-autoscaler/metrics/metrics.go]: Changed to group_type.

**Pull request merged**